src/ObjectPool/src/LeakTrackingObjectPool.cs

```diff
@@ -16,6 +16,7 @@ namespace Microsoft.Extensions.ObjectPool;
 /// </para>
 /// </summary>
 /// <typeparam name="T">The type of object which is being pooled.</typeparam>
+[Obsolete("LeakTrackingObjectPool<T> was only intended for use in diagnostic builds and may be removed in a future release.")]
```

Comment:
The text "was only intended for use in diagnostic builds" can be interpreted as referring to the user's diag builds, not Microsoft's builds. 

I'd phrase it like this, perhaps:

> "LeakTrackingObjectPool<T> is only functional in diagnostic builds of the .NET CLR and EE. It does not function in GA/release builds of .NET used by the general public. This type may be removed in a future release."

Reply:
Thanks @Jehoel. It doesn't actually need a special CLR build to work, but I understand your feedback. How about:

> `LeakTrackingObjectPool<T>` was only intended for use in diagnostic builds of ASP.NET Core. It does not function in any publicly shipped ASP.NET Core versions and may be removed in a future release.

Reply:
👍